c2rust-analyze/scripts/auto_fix_errors.py

```diff
@@ -0,0 +1,286 @@
+import argparse
+from collections import namedtuple
+import json
+import re
+import sys
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description='automatically apply rustc-suggested fixes for compile errors')
+    parser.add_argument('-n', '--dry-run', action='store_true',
+        help="print fixes that would be applied, but don't modify any files")
+    parser.add_argument('path', metavar='ERRORS.JSON',
+        help='output of `rustc --error-format json`')
+    return parser.parse_args()
+
+Fix = namedtuple('Fix', (
+    # Path to the file to modify.
+    'file_path',
+    # Line number where the fix will be applied.  This is used for debug output
+    # only.
+    'line_number',
+    # Start of the byte range to replace.
+    'start_byte',
+    # End (exclusive) of the byte range to replace.
+    'end_byte',
+    # Replacement text (as a `str`, not `bytes`).
+    'new_text',
+    # The original error message.  This is used for debug output only.
+    'message',
+))
+
+LifetimeBound = namedtuple('LifetimeBound', (
+    'file_path',
+    'line_number',
+    # Byte offset of the start of the lifetime parameter declaration.
+    'start_byte',
+    # Byte offset of the end of the lifetime parameter declaration.
+    'end_byte',
+    # The lifetime to use in the new bound.  If `'a: 'b` is the suggested
+    # bound, then `start/end_byte` points to the declaration of `'a`, and
+    # `bound_lifetime` is the string `"'b"`.
+    'bound_lifetime',
+))
+
+RemoveDeriveCopy = namedtuple('RemoveDeriveCopy', (
+    'file_path',
+    'line_number',
+    # Byte offset of the start of the token `Copy`.
+    'start_byte',
+    # Byte offset of the end of the token `Copy`.
+    'end_byte',
+))
+
+MSG_LIFETIME_BOUND = 'lifetime may not live long enough'
```

Comment:
When I ran the analyzer on amalgamated lighttpd, I got a few other common errors:
* `error[E0106]: missing lifetime specifiers`
* ```error[E0277]: `*mut i8` cannot be shared between threads safely``` caused by rewriting some `static mut` globals into `static`.
* ```error[E0392]: parameter `'h16` is never used``` and more like it; basically just unused lifetimes
* `error[E0726]: implicit elided lifetime not allowed here`

Some or all of these could also be handled by your tool (E0392 specifically comes to mind). Should we add them to this PR, or a follow-up?

Reply:
Reverse-engineering a correct fix from the error json is nontrivial, so handling more errors should be left to a separate PR.